pages/docs/styled-system/utility-hooks/use-disclosure.mdx

```diff
@@ -1,60 +1,113 @@
+| `onToggle`           | `function` |         | Callback function to toggle the value of the `isOpen` parameter.                          |
+| `getDisclosureProps` | `function` |         | Callback function to retrieve a set of props for the controlled component.                |
+| `getButtonProps`     | `function` |         | Callback function to retrieve a set of props for the button that triggers the disclosure. |
+
+## Usage
+
+For simple disclosures, you can use the `getDisclosureProps` and
+`getButtonProps` methods returned by the hook to provide the needed attributes
+and handlers to the respective component and button for basic toggling behavior
+and accessibility.
+
+The component that uses `getDisclosureProps` receives the following props:
+
+- An `id` (provided optionally as a custom prop to the hook).
+- A dynamically rendered `hidden` attribute.
+
+`getDisclosureProps` can directly accept any additional props for the component.
+
+The button that uses `getButtonProps` for toggling receives the following props:
+
+- A dynamically rendered `aria-expanded` attribute to let a screen reader know
+  whether the disclosure component is visible.
+- The `aria-controls` attribute using the `id` (provided optionally as a custom
+  prop to the hook). This lets a screen reader know which component is
+  controlled by the button.
+- An onClick handler that uses the `onToggle` callback along with any other
+  click events passed as an `onClick` prop to `getButtonProps`;
+
+`getButtonProps` can also directly accept any additional props for the button.
+
+```jsx
+function Basic() {
+  const { getDisclosureProps, getButtonProps } = useDisclosure()
+
+  const buttonProps = getButtonProps()
+  const disclosureProps = getDisclosureProps()
+  return (
+    <>
+      <Button {...buttonProps}>Toggle Me</Button>
+      <Text {...disclosureProps} mt={4}>
+        This text is being visibly toggled hidden and shown by the button.
+        <br />
+        (Inspect these components to see the rendered attributes)
+      </Text>
+    </>
+  )
+}
+```
+
+If you need more control over the disclosure behavior, you can use a combination
+of the getter props along with the other methods and values returned by the
+hook.
+
+Below is use of the return fields of the hook without a getter to control the
+`Drawer` component on button toggle.
+
+```jsx
+function Example() {
+  const { isOpen, onOpen, onClose } = useDisclosure()
+
+  return (
+    <>
+      <Button onClick={onOpen}>Open Drawer</Button>
+      <Drawer placement='right' onClose={onClose} isOpen={isOpen}>
+        <DrawerOverlay />
+        <DrawerContent>
+          <DrawerHeader borderBottomWidth='1px'>Basic Drawer</DrawerHeader>
+          <DrawerBody>
+            <p>Some contents...</p>
+            <p>Some contents...</p>
+            <p>Some contents...</p>
+          </DrawerBody>
+        </DrawerContent>
+      </Drawer>
+    </>
+  )
+}

```

Comment:
I'd move this above the other example. This is the more basic way to use useDisclosure. The example you added is more advanced usage. Maybe  2 sub sections for usage could make sense: Basic and Advanced

Reply:
Oh! I figured because of only using the getters were the basic way because they do a simple toggle and handle needed accessibility, and you only specify a button for toggle and a component that is to be visibly toggled. I would like to understand why it is considered the advanced approach! 😄 

Aside from that, I will go ahead and make the change. 👍🏼

Reply:
That's is just my opinion. I think so because in the more 'basic' usage you don't bother about anything except open/close something. We don't have to put it explicitly in sub sections, that might cause friction indeed.

Reply:
Ok! I'll just flip the examples and tweak the wording of the details.

Reply:
@noobinthisgame good to check this again. 😃